src/regdesk/communication_manager.js

```diff
@@ -42,26 +46,26 @@ export class CommunicationManager {
    * @param {string} searchString - The string to search for.
    * @return {Promise<RegistrantDetails[]>} The promise that will return the search results.
    */
-  startSearchForRegByName(searchString) {
-    // Start the search process, internally storing the promise and handling
-    // the return value. Something like:
-    // this.#searhPromise = this
-    //   .whateverCommunicationThing.actuallyDoTheSearch(searchString)
-    //   .then(handleSearchResults);
-
-    // We then expect the LoadingState to add its own .then to the promise and
-    // handle it appropriately.
-
-    // In the meantime, mock it out
-    const arr = [];
-    const len = Math.floor(Math.random() * (4));
-    for (let i = 0; i < len; i++) {
-      arr.push(this.#testGetRandomReg());
-    }
-
-    return new Promise((resolve) => setTimeout(resolve, 500))
-      .then(() => arr) // The mock one!
-      .then(this.handleSearchResults.bind(this)); // The real one!
+  async startSearchForRegByName(searchString) {
+    // FIXME I think we still need to do some client side filtering for cancelled/duplicate/etc memberships.
+    const bearerToken = await this.#loginMgr.getBearerToken();
+    return RegfoxApi.searchRegistrations(searchString, bearerToken)
+      .then((result) => result.registrants.slice(0, 10)) // The next call hammers regfox, so we want to limit this as much as possible
+      .then((results) => Promise.all(results.map((result) => RegfoxApi.getRegistrationInfo(result.id, bearerToken))))
```

Comment:
I guess there really isn't a lot of useful information to filter down first eh? Are we just generally hoping that the majority of searches looks for more specific results than 10 registrations?

Reply:
Fair point, I think at the moment since we have no mechanism to inform the user "please search more specifically" then we can't include something like this.